src/EnergyPlus/DataConvergParams.hh

```diff
@@ -141,73 +141,45 @@ namespace DataConvergParams {
-        Array1D<Real64> HVACEnthalpySupplyDeck2ToDemandTolValue; // Queue of convergence "results"
-        Array1D_bool HVACPressureNotConverged;                   // Flag to show energy convergence   or failure
-        Array1D<Real64> HVACPressureDemandToSupplyTolValue;      // Queue of convergence "results"
-        Array1D<Real64> HVACPressureSupplyDeck1ToDemandTolValue; // Queue of convergence "results"
-        Array1D<Real64> HVACPressueSupplyDeck2ToDemandTolValue;  // Queue of convergence "results"
-        Array1D_bool HVACQualityNotConverged;                    // Flag to show energy convergence   or failure
-        Array1D<Real64> HVACQualityDemandToSupplyTolValue;       // Queue of convergence "results"
-        Array1D<Real64> HVACQualitSupplyDeck1ToDemandTolValue;   // Queue of convergence "results"
-        Array1D<Real64> HVACQualitySupplyDeck2ToDemandTolValue;  // Queue of convergence "results"
-
-        // Default Constructor
-        HVACAirLoopIterationConvergenceStruct()
-            : HVACMassFlowNotConverged(3, false), HVACFlowDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACFlowSupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACFlowSupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0),
-              HVACHumRatNotConverged(3, false), HVACHumDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACHumSupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACHumSupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0),
-              HVACTempNotConverged(3, false), HVACTempDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACTempSupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACTempSupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0),
-              HVACEnergyNotConverged(3, false), HVACEnergyDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACEnergySupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACEnergySupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0),
-              HVACEnthalpyNotConverged(3, false), HVACEnthalpyDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACEnthalpySupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACEnthalpySupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0),
-              HVACPressureNotConverged(3, false), HVACPressureDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACPressureSupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACPressueSupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0),
-              HVACQualityNotConverged(3, false), HVACQualityDemandToSupplyTolValue(ConvergLogStackDepth, 0.0),
-              HVACQualitSupplyDeck1ToDemandTolValue(ConvergLogStackDepth, 0.0), HVACQualitySupplyDeck2ToDemandTolValue(ConvergLogStackDepth, 0.0)
-        {
-        }
+        std::array<bool, 3> HVACMassFlowNotConverged = {false};                   // Flag to show mass flow convergence
```

Comment:
All these Array1D's have a known size at compile time, so converted them all to std::array of the right length.

Reply:
This is awesome, but you are missing an opportunity for additional structuring.

```cpp
struct Whatever {
     Real64 DemandToSupplyTolValue, SupplyDeck1ToDemandTolValue, SupplyDeck2ToDemandTolValue;
     bool NotConverged;

     Whatever() : DemandToSupplyTolValue(0.0), SupplyDeck1ToDemandTolValue(0.0), SupplyDeck2ToDemandTolValue(0.0), NotConverged(false) {}
}

std::array<struct Whatever, ConvergLogStackDepth> HVACFlow;
std::array<struct Whatever, ConvergLogStackDepth> HVACHum;
std::array<struct Whatever, ConvergLogStackDepth> HVACTemp;

...
```

Reply:
I held off on this one piece for now to keep my next commit reasonably tractable, but I could still do this here, or in a later effort.